File: bin/bin/old_bin/bin/math_0004.c
#include <stdio.h>
#include<string.h>

//how many digits does this number have 1234567891011....201520162017

void main(void){
	int count, result = 0;
//	for(int i=1; i <= 2017; i++){
//		if (i < 10) count = 1; else
//		if (i < 100) count = 2; else
//		if (i < 1000) count = 3; else
//		if (i < 10000) count = 4;
//		result += count;		
//	}
	result = 2017*4 - 999 - 99 - 9 ;
	printf("result: %d\n", result);
}

File: bin/old_bin/bin/math_0004.py
#!/usr/bin/python
import math
#How many digits does this number have 1234567891011...201620172018

class exercise :
    def doit(self):
        count=0
	num_digit=0
        for a in range(1, 2018):
		if (a < 10):
			num_digit=1
		elif (a < 100):
			num_digit=2
		elif (a < 1000):
			num_digit=3
		elif (a < 10000):
			num_digit=4
		count += num_digit
#		if (a == 2017):
#			print('num_digit={}, count={}').format(num_digit, count)
        print('count is {}').format(count)


if __name__ == '__main__':	
	x = exercise()
	x.doit()

